src/Reflection/DeprecatableReflection.php

```diff
@@ -0,0 +1,10 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection;
+
+interface DeprecatableReflection
+{
+
+	public function isDeprecated(): bool;
```

Comment:
I don't understand why this is a separate interface, why don't just add the method to all the interfaces?

Reply:
Some things cannot be marked as deprecated. Take a method defined in a PHPDoc for example:

```
/**
 * @method void foo()
 */
class Foo {}
```

Thus one has to check if deprecation is supported first via `$method instanceof DeprecatableReflection` which is cleaner than `method_exists($method, 'isDeprecated')`. One could explicitly mark all non-deprecatable structures as not deprecated though.